api-metastore/src/main/java/org/zalando/nakadi/controller/EventTypeController.java

```diff
@@ -144,7 +144,8 @@ private HttpHeaders generateWarningHeaders(final EventTypeBase eventType) {
 
         final List<String> warnings = Lists.newArrayList(nakadiSettings.getWarnAllDataAccessMessage());
 
-        if (eventType.getCleanupPolicy().equals(CleanupPolicy.COMPACT)) {
+        if (eventType.getCleanupPolicy().equals(CleanupPolicy.COMPACT) ||
+                eventType.getCleanupPolicy().equals(CleanupPolicy.COMPACT_AND_DELETE)) {
```

Comment:
Do you think we can skip warning for `compact_and_delete` event-types? If I understand correctly, the data would be always deleted too for `compact,delete` event-types, following retention time policies.

Another approach here might be to just change the internal warn message.

Reply:
true a667c1397